functions/src/api/v1/users/getUser.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response } from 'express'
+import { ForbiddenError } from '../../errors/HttpError'
+import { getUserDb } from '../../../db/getUserDb'
+
+/**
+ * @swagger
+ * /v1/users/{userId}:
+ *   get:
+ *     description: Get user details, subscription and quota status. This will not send any payment details or API keys.
+ *     tags: ['users']
+ *     produces:
+ *       - application/json
+ *     responses:
+ *       200:
+ *         description: User details
+ *         content:
+ *           application/json:
+ *             example: {
+ *               id: 'vXIUuaGkH4kukbwRH5cU',
+ *               updatedAt: 1605969562,
+ *               subscription: {
+ *                   isActive: true,
+ *                   isQuotaReached: false
+ *               },
+ *               sentEmails: 1,
+ *               sentSMSs: 2
+ *             }
+ *             schema:
+ *               type: object
+ *               $ref: '#/components/schemas/User'
+ *       401:
+ *         description: missing authorization bearer token
+ *       403:
+ *         description: authorization header present but not valid
+ *       412:
+ *         description: authorization header present but not formatted correctly
+ */
+export const getUser = async (req: Request, res: Response) => {
+    const currentUid = res.locals.uid
+    const requestedUserId = req.params.userId
+
+    if (requestedUserId !== currentUid) {
+        throw new ForbiddenError('Not authorized to get this resource')
+    }
+
+    const user = await getUserDb(currentUid)
+
+    res.send({
+        user: {
```

Comment:
as a general rule, I think it only adds extra complexity to wrap responses with a key, we could directly send back the object here, since it's clear what `getUser` should return

Reply:
it seems simplie yes, but what if we want to add some extra field to the response later on? We will have to brake compatibility, make API v2 and so on. here it seems to be a good tradeoff.

Reply:
I see your point, but I've never seen this practice in my experience. If a `getUser` function, we may add some user related extra fields, which would still be part of the `User` object, so I still think wrapping all API responses  adds complexity. @ntchou ?